src/nunit.analyzers.tests/TestCaseUsage/TestCaseUsageAnalyzerTests.cs

```diff
@@ -239,6 +239,85 @@ class AnalyzeArgumentHasImplicitConversion
             AnalyzerAssert.Valid<TestCaseUsageAnalyzer>(testCode);
         }
 
+        [Test]
+        public void AnalyzeArgumentIsStringConvertedToTypeWithCustomConverter()
+        {
+            var testCode = TestUtility.WrapClassInNamespaceAndAddUsing(@"
+    using System.ComponentModel;
+
+    class AnalyzeArgumentIsStringConvertedToEnum
+    {
+        [TestCase(""A"")]
+        public void Test(CustomType p) { }
+    }
+
+    [TypeConverter(typeof(CustomTypeConverter))]
+    struct CustomType { }
+    class CustomTypeConverter : TypeConverter { }");
+
+            AnalyzerAssert.Valid<TestCaseUsageAnalyzer>(testCode);
+        }
+
+        [Test]
+        public void AnalyzeArgumentIsNullConvertedToTypeWithCustomConverter()
+        {
+            var testCode = TestUtility.WrapClassInNamespaceAndAddUsing(@"
+    using System.ComponentModel;
+
+    class AnalyzeArgumentIsStringConvertedToEnum
+    {
+        [TestCase(null)]
+        public void Test(CustomType p) { }
+    }
+
+    [TypeConverter(typeof(CustomTypeConverter))]
+    struct CustomType { }
+    class CustomTypeConverter : TypeConverter { }");
+
+            AnalyzerAssert.Valid<TestCaseUsageAnalyzer>(testCode);
+        }
+
+        [Test]
+        public void AnalyzeArgumentIsStringConvertedToTypeWithCustomConverterOnBaseType()
+        {
+            var testCode = TestUtility.WrapClassInNamespaceAndAddUsing(@"
+    using System.ComponentModel;
+
+    class AnalyzeArgumentIsStringConvertedToEnum
+    {
+        [TestCase(""A"")]
+        public void Test(CustomType p) { }
+    }
+
+    class CustomType : BaseType { }
+    [TypeConverter(typeof(BaseTypeConverter))]
+    class BaseType { }
+    class BaseTypeConverter : TypeConverter { }");
+
+            AnalyzerAssert.Valid<TestCaseUsageAnalyzer>(testCode);
+        }
+
+        [Test]
+        public void AnalyzeArgumentIsIntConvertedToTypeWithCustomConverter()
+        {
+            var testCode = TestUtility.WrapClassInNamespaceAndAddUsing(@"
+    using System.ComponentModel;
+
+    class AnalyzeArgumentIsStringConvertedToEnum
```

Comment:
```suggestion
    class AnalyzeArgumentIsIntConvertedToTypeWithCustomConverter
```

Reply:
Oh, thanks. Is it Okay if I call it `C` like I usually do?

Reply:
Sure. When reviewing this I was also consider what we gained from this (except for additional work 😄).